### DISCUSSION TITLE
[Bug]: Switching from Dongle to RS485 causes grid consumption

### DISCUSSION THREAD

**Discussion by JohnMcLear** [wlcrs/huawei_solar#460] [General]
### Describe the issue

This is probably not an issue with this software but it's worth documenting here.

Background: Three inverters(each with 2 batteries) 6K-L2, all connected historically over WiFi using the SDongle.  Lots of reliability issues so I bit the bullet and went Elfin EW11 and setup as per docs, this means I'm using RS485.

With the EW11 running and reporting all inverters/batteries everything looked great until I noticed a problem..  Since dropping the SDongle and using the EW11 my inverters have not been pulling from the batteries (which are at 100%) and instead have been pulling energy from the Grid!  I noticed that in HA the master inverter is showing as "Power limited", I directly connected to this inverter and checked the config and it looks right.  No alarms or errors or anything that suggest why it's power limited..

I can see active power, grid current etc etc. in the inverter so it's seeing the smart energy meter thingy.  For now I've put the SDongle in so my grid consumption isn't high and I will keep an eye on usage.

Questions..

1. Why would removing the SDongle and using RS485 change the config?
2. Any tips for checking the config IE where I might want to look to see if something has changed?  I checked maximum-self consumption, allowable feed in amounts etc.

Anything else you can think to check or suggestions for what I'm doing wrong..

### Bescribe your Huawei Solar Setup

Inverter Type: 6K-L2
Inverter Firmware version: n/a
SDongle present: not any more 
Power meter present: single phase
Battery: LUNA2000
Battery Firmware version: n/a

### How do you connect to the inverter?

Over serial, with a serial-to-WiFi stick

### Upload your Diagnostics File

N/a

### Upload your relevant debug logs

```text
n/a
```

### Please confirm the following:

- [X] The problem is still present in the latest release of this integration.
- [x] I did not find an existing issue describing this problem.
- [ ] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by wlcrs** [comment]
The integration does not change any settings on the inverter by itself. It will only perform a Modbus write operation when the user explicitly changes a configurable entity in HA.

I'm not intimately familiar with the various settings that Huawei offers. I also don't know what that causes that "Power limited" status, or how to resolve it.

**Comment by Roving-Ronin** [comment]
Wondering why your removing the USB WiFi/FE dongle, as for the system to be able to still connect to Solar Fusion portal would require the primary inverter to have a dongle installed.

To confirm you've got all the inverters daisy chained using RS485 from primary --> secondary --> tertiary ?

As per Huawei Wiring Diagram:
![SUN2000-6KTL-M1 + LUNA2000 ESS Wiring Diagram](https://github.com/wlcrs/huawei_solar/assets/108674933/9fdf72c9-722c-40b0-9acd-f80a97855651)

as shown here, Pin 1 (RS485A1-1) and Pin 3 (RS485B1-1) are the pins used to connect 'downstream', whilst Pin 2 (RS485A1-2) and Pin 4 (RS485B1-2) are then used to terminate the connection from the 'upstream' device/inverter.

**Comment by JohnMcLear** [comment]
@Roving-Ronin see 
> Warning⚠: The SDongleA-05 dongle interferes with devices connected to the RS485A1 and RS485B1 pins on the inverter. If your inverter has such a dongle installed you cannot use a serial connection option. We recommend connecting to the Modbus-TCP port exposed by that dongle instead." -- https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter#overview

That's why you have to drop the SDongle.

**Comment by Roving-Ronin** [comment]
D'Oh was thinking I had my USB dongle in, whilst SmartLogger was connected... just realised it was the inbuilt WiFi / AP that I was attempting to bridge with.. hence the RS485 wasn't affected. 
Noticed the Connectivity page only has reference to the L1 COM port, have updated with the COM port (pin allocations) for the M3 models.
